vehicle/Car.h

```diff
@@ -9,21 +9,21 @@
 // Car 
 class Car {
 public:
-    Car(IEngine* engine, IGear* gear)
-        : engine(engine), gear(gear) {}
+    Car(IEngine* m_engine, IGear* m_gear)
```

Comment:
why do you use the same names for the parameters and class members?

Reply:
updated, p_ for parameters, m_ for class members